cpmpy/tools/xcsp3/xcsp3_natives.py

```diff
@@ -0,0 +1,50 @@
+import cpmpy as cp
+import numpy as np
+
+# --------------------------------- OR-Tools --------------------------------- #
+
+def ort_nooverlap2d(cpm_ortools, cpm_expr):
+    start_x, dur_x, end_x, start_y, dur_y, end_y = cpm_ortools.solver_vars(cpm_expr.args)
+    intervals_x = [cpm_ortools.ort_model.NewIntervalVar(s,d,e, f"xinterval_{s}-{d}-{d}") for s,d,e in zip(start_x,dur_x,end_x)]
+    intervals_y = [cpm_ortools.ort_model.NewIntervalVar(s,d,e, f"yinterval_{s}-{d}-{d}") for s,d,e in zip(start_y,dur_y,end_y)]
+    return cpm_ortools.ort_model.add_no_overlap_2d(intervals_x, intervals_y)
+
+def ort_subcircuit(cpm_ortools, cpm_expr):
+    x = cpm_expr.args
+    N = len(x)
+    arcvars = cp.boolvar(shape=(N,N))
+    # post channeling constraints from int to bool
+    cpm_ortools += [b == (x[i] == j) for (i,j),b in np.ndenumerate(arcvars)]
+    # post the global constraint
+    # posting arcs on diagonal (i==j) allows for subcircuits
+    ort_arcs = [(i,j,cpm_ortools.solver_var(b)) for (i,j),b in np.ndenumerate(arcvars)] # Allows for empty subcircuits
+    return cpm_ortools.ort_model.AddCircuit(ort_arcs)
+
+def ort_subcircuitwithstart(cpm_ortools, cpm_expr):
+    x = cpm_expr.args
+    N = len(x)
+    arcvars = cp.boolvar(shape=(N,N))
+    # post channeling constraints from int to bool
+    cpm_ortools += [b == (x[i] == j) for (i,j),b in np.ndenumerate(arcvars)]
+    # post the global constraint
+    # posting arcs on diagonal (i==j) allows for subcircuits
+    ort_arcs = [(i,j,cpm_ortools.solver_var(b)) for (i,j),b in np.ndenumerate(arcvars) if not ((i == j) and (i == cpm_expr.start_index))] # The start index cannot self loop and thus must be part of the subcircuit.
+    return cpm_ortools.ort_model.AddCircuit(ort_arcs)
```

Comment:
similar to above
(not that in each case, you will need to add the variables etc directly to the cpmpy model, and then change the `return cpm_ort.ort_mod.AddCircuit(ort_arcs)` by m += DirectConstraint("AddCircuit", ort_arcs)